src/HypLSP8.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.19;
+
+import { TokenRouter } from "@hyperlane-xyz/core/contracts/token/libs/TokenRouter.sol";
+
+import { LSP8IdentifiableDigitalAssetInitAbstract } from
+    "@lukso/lsp8-contracts/contracts/LSP8IdentifiableDigitalAssetInitAbstract.sol";
+
+import { _LSP4_TOKEN_TYPE_TOKEN } from "@lukso/lsp4-contracts/contracts/LSP4Constants.sol";
+
+import { _LSP8_TOKENID_FORMAT_NUMBER } from "@lukso/lsp8-contracts/contracts/LSP8Constants.sol";
+
+/**
+ * @title Hyperlane LSP8 Token Router that extends LSP8 with remote transfer functionality.
+ * @author Abacus Works
+ */
+contract HypLSP8 is LSP8IdentifiableDigitalAssetInitAbstract, TokenRouter {
+    constructor(address _mailbox) TokenRouter(_mailbox) { }
+
+    /**
+     * @notice Initializes the Hyperlane router, LSP8 metadata, and mints initial supply to deployer.
+     * @param _mintAmount The amount of NFTs to mint to `msg.sender`.
+     * @param _name The name of the token.
+     * @param _symbol The symbol of the token.
+     */
+    function initialize(uint256 _mintAmount, string memory _name, string memory _symbol) external initializer {
```

Comment:
This function is missing the initialization with the Mailbox.

https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/7e9e248beff122e7f1093eb8883b4ffe9f04a919/solidity/contracts/token/HypERC721.sol#L30C1-L34

<img width="857" alt="image" src="https://github.com/user-attachments/assets/1a829e3d-f2a4-4863-92aa-68cd0c5c0f86">